source/socketstuff.lua

```diff
@@ -11,183 +11,192 @@ gbolIsAHost = false                					-- defaults to NOT a host until the play
-		ss.AddItemToHostOutgoingQueue(msg)
-		ss.SendToClients()
+		ss.addItemToHostOutgoingQueue(msg)
+		ss.sendToClients()
 		msg = {}
 	end
 
 Put the following code into love.update so that clients can do client things:
 
 	if gbolIsAClient then
-		ss.ClientListenPort()
-
+		ss.clientListenPort()
+		
 		-- get just one item from the queue and process it
-		local incoming = ss.GetItemInClientQueue()		-- could be nil
+		local incoming = ss.getItemInClientQueue()		-- could be nil
 		if incoming ~= nil then
 			print(inspect(msg))
 		end
 
 		msg = whatever 		-- string, number or table.
-		ss.AddItemToClientOutgoingQueue(msg)	--
-		ss.SendToHost()
+		ss.addItemToClientOutgoingQueue(msg)	-- 
+		ss.sendToHost()
 		msg = {}
 	end
-
+	
 ]]
 
 
 
 local socketstuff = {}
 
-local arrHostIncomingQueue = {}
-local arrClientIncomingQueue = {}
-local arrHostOutgoingQueue = {}
-local arrClientOutgoingQueue = {}
-local arrClientNodes = {}
+local hostIncomingQueue = {}
+local clientIncomingQueue = {}
+local hostOutgoingQueue = {}
+local clientOutgoingQueue = {}
+local cilentNodes = {}
 
-udpclient = nil
-udphost = nil
+local udpClient = nil
+local udpHost = nil
 
-function socketstuff.HostListenPort()
+function socketstuff.hostListenPort()
 -- listens for a message and adds it to the queue
-    local data, ip, port = udphost:receivefrom()
+    local data, ip, port = udpHost:receivefrom()
 	local unpackeddata
     if data then
 		unpackeddata = bitser.loads(data)
-        table.insert(arrHostIncomingQueue,unpackeddata)
+        table.insert(hostIncomingQueue,unpackeddata)
     end
-    --socket.sleep(0.01)    -- this doesn't seem to do much so I removed it
-
+    -- socket.sleep(0.01)    -- this doesn't seem to do much so I removed it
+	
 	local node = {}
     node.ip = ip
     node.port = port
-
+	
 	if port == nil or unpackeddata == nil then
 		-- no message, do nothing
 	else
 		local bolAddClient = true
-		for k,v in pairs(arrClientNodes) do
+		for k,v in pairs(cilentNodes) do
 			if node.ip == v.ip and node.port == v.port then
 				-- this node is already captured
 				bolAddClient = false
 				break
 			end
 		end
 		if bolAddClient then
-			table.insert(arrClientNodes,node)
+			table.insert(cilentNodes,node)
 		end
 	end
 
 end
 
-function socketstuff.ClientListenPort()
-    local data, msg = udpclient:receive()
+
+function socketstuff.clientListenPort()
+    local data, msg = udpClient:receive()
 
 	if data then
 		local unpackeddata = bitser.loads(data)
-        table.insert(arrClientIncomingQueue,unpackeddata)
+        table.insert(clientIncomingQueue,unpackeddata)
     end
 end
 
-function socketstuff.GetItemInHostQueue()
+
+function socketstuff.getItemInHostQueue()
 -- returns the first/oldest item in the message queue
 
 	local retval
-	if #arrHostIncomingQueue > 0 then
-		retvalue = arrHostIncomingQueue[1]
-		table.remove(arrHostIncomingQueue,1)
+	if #hostIncomingQueue > 0 then
+		retvalue = hostIncomingQueue[1]

```

Comment:
This is a global variable by accident because the above predefined local is called differently. Also it should be named `retValue` or even better just `client` or `incomingClient`.

And maybe to optimize the function even more:
```
function socketstuff.getItemInClientQueue()
-- returns the first/oldest item in the message queue and deletes that item from the queue
	if #clientIncomingQueue > 0 then
		local client = clientIncomingQueue[1]
		table.remove(clientIncomingQueue,1)
		return client
	end
end
```
Since the return for failure (no more clients in queue) is nil anyway.

Reply:
There needs to be a RETURN statement if #table < 1 but yes - there is some room to optimise. Thanks.